framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2157,7 +2173,8 @@ protected void checkMethodInvocability(AnnotatedExecutableType method,
 
         treeReceiver.addAnnotations(rcv.getEffectiveAnnotations());
 
-        if (!atypeFactory.getTypeHierarchy().isSubtype(treeReceiver, methodReceiver)) {
+        if (!skipReceiverSubtypeCheck(node, methodReceiver, rcv) &&
```

Comment:
What is your reasoning for add this "skipReceiverSubtypeCheck" method rather than just overriding checkMethodInvoabilty?  Also, it would be good if skipReceiverSubtypeCheck had a JavaDoc explaining its purpose.  Maybe checkMethodIvnv's javadoc should mention skipReceiverSubtypeCheck as an extension point.